assets/src/design-system/components/search/input/components.js

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+/**
+ * Internal dependencies
+ */
+import { Chevron, Close as Clear, Search } from '../../../icons';
+import { themeHelpers, THEME_CONSTANTS } from '../../../theme';
+import { Z_INDEX } from '../constants';
+
+export const InputContainer = styled.div(
+  ({ theme, alignCenter }) => css`
+    display: flex;
+    position: relative;
+    box-sizing: border-box;
+    height: 36px;
+    width: 100%;
+    max-width: 276px;
+    background-color: ${theme.colors.bg.primary};
+    color: ${theme.colors.fg.primary};
+
+    input {
+      padding: 8px 20px 8px ${alignCenter ? 'calc(50% - 8px)' : '32px'};
```

Comment:
`alignCenter` is tied to `isOpen` which makes the input rerender and lose any keyboard class for focus. This fixes it!